index.js

```diff
@@ -533,12 +539,14 @@ function parse (args, opts) {
     var type = 'boolean'
 
     if (flags.strings && flags.strings[key]) type = 'string'
+    else if (flags.numbers && flags.numbers[key]) type = 'number'
     else if (flags.arrays && flags.arrays[key]) type = 'array'
 
     return type
   }
 
   function isNumber (x) {
+    if (flags.numbers && flags.numbers[x] && !isNaN(x)) return true
```

Comment:
@bcoe got this going so far: 
```js
    if (!checkAllAliases(key, flags.strings)) {
      if (isNumber(val) || checkAllAliases(key, flags.numbers)) value = Number(val)
      if (!isNumber(val) && checkAllAliases(key, flags.numbers)) value = NaN
    }
```

The issue is that `isNumber(undefined)` returns false. I am wondering if we ought to change that to `return true` instead, or try another work around?